### DISCUSSION TITLE
Combining revisions from two sources

### DISCUSSION THREAD

**Discussion by chepati** [argoproj/argo-cd#20344] [General]
We have an application that deploys a helm chart from source A and values file from source B.  The helm chart's application has its own version (let's call it revisionA) which is half of what we want.  The other half is the sha1 of the git commit (let's call it revisionB) of the values file from source B.

We want to concatenate the two revisions into a string "revisionA-revisionB" and have ArgoCD add it as a label (or annotation) to every pod this helm chart launches.

Any suggestions how to do this?
Thanks.

**Comment by jbbdpl** [comment]
I would like to add a context here

We are using latest (close to latest) ArgoCD v2.12.4. Multiple sources tab was introduce. We do have source A for helm and source B for values. We wanted to pass ArgoCD build variable into our pod - ARGOCD_APP_REVISION
https://argo-cd.readthedocs.io/en/stable/user-guide/build-environment/

What it seems to be in logic of ArgoCD
https://github.com/argoproj/argo-cd/blob/master/reposerver/repository/repository.go#L408
when 2 revisions are in use - he uses helm revision instead of sha 

Can this be looked into and some solution be proposed?

**Reply by chepati** [reply]
Ping.

Perhaps I can try to elaborate further.

We have an application with two sources:

sources:
  - repoURL: git@git.repo/repo.git
    targetRevision: main
    path: path/to/helm/values.yaml
    ref: helmValues
  - repoURL: https://helm.chart.url
    chart: chart-name
    targetRevision: 1.0.0
    helm:
      valueFiles:
        - $helmValues/path/to/helm/values.yaml
        
 This is all fine -- my helm chart deploys with the values.yaml from the git repo.  However, $ARGOCD_APP_REVISION always reports 1.0.0, which is the helm chart version.  What I want is to either choose which revision I want (the helm chart or the git commit sha of the git repo where values.yaml is) or to be able to access the two sources' revision as in $ARGOCD_APP_REVISION[0] and $ARGOCD_APP_REVISION[1].
 
 We ended up using an umbrella chart deployment, but I want to avoid it, if I can.
 
 Ideas, suggestions?

**Comment by cggittraining** [comment]
We have the same issue
Argo version 2.13.3

multi source:
    - chart: chart-name
      repoURL: git@git.repo/repo.git
      targetRevision: 2.0.17          # chart version — same for all envs  
      helm:  
        parameters:  
          - name: image.tag  
            value: $ARGOCD_APP_SOURCE_0_REVISION #0.1.0.2998.41843b02  
        valueFiles:  
          - $values/src/main/helm/values/values.yaml  
          - $values/src/main/helm/values/{{ .helmValuesFile }}
          - $values/src/main/helm/secrets/{{ .secretFile }} 
    - repoURL: https://helm.chart.url/
      targetRevision: '{{ .targetRevision }}'
      ref: values  
      
 and results only from Chart.yaml
 
 apiVersion: v2
name: "template.name"
description: "Helm Chart for ... services"
type: "application"
version: 1.0.0
appVersion: "1.0.0"

 template was published with version 2.0.17 in harbor (oci repo)
 
 ARGOCD_APP_REVISION -> 2.0.17
ARGOCD_APP_SOURCE_0_REVISION -> 1.0.0
ARGOCD_APP_SOURCE_1_REVISION -> 1.0.0
ARGOCD_APP_SOURCE_2_REVISION -> 1.0.0
ARGOCD_APP_REVISION_SHORT -> 2.0.17
